services/server.service.ts

```diff
@@ -329,6 +333,36 @@ export async function getAllProfilesForServer(
     .returns<ServerUserProfile>();
 }
 
-type GetAllProfilesForServerResponse = Awaited<ReturnType<typeof getAllProfilesForServer>>;
-export type GetAllProfilesForServerResponseSuccess = GetAllProfilesForServerResponse['data'];
-export type GetAllProfilesForServerResponseError = GetAllProfilesForServerResponse['error'];
+type GetAllProfilesForServerResponse = Awaited<
+  ReturnType<typeof getAllProfilesForServer>
+>;
+export type GetAllProfilesForServerResponseSuccess =
+  GetAllProfilesForServerResponse['data'];
+export type GetAllProfilesForServerResponseError =
+  GetAllProfilesForServerResponse['error'];
+
+export async function updateServerIcon(
+  supabase: SupabaseClient<Database>,
+  filePath: string,
+  image: any,
+  server_id: number
+) {
+  const { data, error } = await supabase.storage
+    .from('servericons')
+    .update(filePath, image, { upsert: true, cacheControl: '1' });
+
+  const publicURL = supabase.storage
+    .from('servericons')
+    .getPublicUrl(data!.path);
```

Comment:
On updating server icon, `data` is null here. We should nullcheck this and report an error appropriately on the front